File: niuniu.py
def getNiu(card):    
    sum=0
    for x in range(5):
        sum+=card[x]
    for i in range(5):
        for j in range(i+1, 5):
            for k in range(j+1, 5):
                if (card[i] + card[j] + card[k]) % 10 == 0:
                    return sum%10;
    print "no Niu"
    return -1;
if __name__ == "__main__":
    card = [10,1,2,3,7]
    for i in range(5):
        if card[i]>=10:
            card[i]=0;
    print getNiu(card)
                    
        

File: niuniu.cpp
#include<iostream>
#include<algorithm>
using namespace std;
int main(){
	int card[] = {3,2,1,7,10};
	int sum=0;
	for(int i=0;i<5;i++)
		sum+=card[i];
	do{
    	if((card[0]+card[1]+card[2])%10==0)
    	{
    		cout<<sum%10;
    		break;
    	}
	}
	while (next_permutation(card,card+5));
	return 0;
}
